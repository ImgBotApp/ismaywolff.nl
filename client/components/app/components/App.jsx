import React, { PropTypes } from 'react'
import Router from 'react-router/BrowserRouter'
import { Provider } from 'react-redux'
import Routes from './Routes'

function App({ store }) {
  return (
    <Provider store={store}>
      <Router>
        <Routes />
      </Router>
    </Provider>
  )
}

App.propTypes = {
  store: PropTypes.object.isRequired
}

export default App
