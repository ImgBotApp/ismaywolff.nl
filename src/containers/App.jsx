/* @flow */

/**
 * Dependencies
**/

import React from 'react';
import Match from 'react-router/Match';
import { style } from 'glamor';
import AboutPage from '../components/AboutPage';
import MouldPage from '../components/MouldPage';
import InductionPage from '../components/InductionPage';
import SubjectPage from '../components/SubjectPage';
import EyeOfTheBeholderPage from '../components/EyeOfTheBeholderPage';
import WorkPage from '../components/WorkPage';
import Header from '../components/Header';

/**
 * Styles
**/

const wrapper = style({
  maxWidth: '30rem',
});

/**
 * Component: App
**/

const App = () =>
  (<div {...wrapper}>
    <Header />
    <Match exactly pattern="/" component={WorkPage} />
    <Match pattern="/about" component={AboutPage} />
    <Match pattern="/work/mould" component={MouldPage} />
    <Match pattern="/work/induction" component={InductionPage} />
    <Match pattern="/work/subject" component={SubjectPage} />
    <Match pattern="/work/eye-of-the-beholder" component={EyeOfTheBeholderPage} />
  </div>);

/**
 * Export
**/

export default App;
