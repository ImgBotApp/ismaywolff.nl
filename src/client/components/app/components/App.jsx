import React from 'react'
import { object, func, shape } from 'prop-types'
import { Router } from 'react-router'
import { Provider } from 'react-redux'
import { ScrollToTop } from '../../scroll'
import { Root } from '../../../../shared/components/root'

const App = ({ history, store }) => (
  <Provider store={store}>
    <Router history={history}>
      <div>
        <ScrollToTop />
        <Root />
      </div>
    </Router>
  </Provider>
)

App.propTypes = {
  store: shape({
    dispatch: func.isRequired,
    getState: func.isRequired
  }).isRequired,
  history: shape({
    location: object.isRequired
  }).isRequired
}

export default App
