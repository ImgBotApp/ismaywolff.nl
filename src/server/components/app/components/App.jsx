import React from 'react'
import { string, func, shape, object } from 'prop-types'
import { StaticRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import { Root } from '../../../../shared/components/root'

const App = ({ location, context, store }) =>
  <Provider store={store}>
    <StaticRouter location={location} context={context}>
      <Root />
    </StaticRouter>
  </Provider>

App.propTypes = {
  store: shape({
    dispatch: func.isRequired,
    getState: func.isRequired
  }).isRequired,
  location: string.isRequired,
  // eslint-disable-next-line react/forbid-prop-types
  context: object.isRequired
}

export default App
