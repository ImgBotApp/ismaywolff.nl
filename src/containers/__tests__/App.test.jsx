/* eslint-env jest */
/* eslint-disable import/no-extraneous-dependencies */

import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import App from '../App';

describe('<App />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<App>child</App>);

    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
