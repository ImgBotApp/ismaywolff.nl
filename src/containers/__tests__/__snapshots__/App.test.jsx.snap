exports[`<App /> renders correctly 1`] = `
<div
  data-css-zrm7yy="">
  <Header />
  <Match
    component={[Function]}
    exactly={true}
    pattern="/" />
  <Match
    component={[Function]}
    exactly={false}
    pattern="/about" />
  <Match
    component={[Function]}
    exactly={false}
    pattern="/work/mould" />
  <Match
    component={[Function]}
    exactly={false}
    pattern="/work/induction" />
  <Match
    component={[Function]}
    exactly={false}
    pattern="/work/subject" />
  <Match
    component={[Function]}
    exactly={false}
    pattern="/work/eye-of-the-beholder" />
</div>
`;
